### PR TITLE
br: Adjust the backup organization structure

### LINKED ISSUE #13063: br: failed to backup to s3 due to the limit of request rate.
Dup: pingcap/tidb#30087

### PR DESCRIPTION
**Pull request by MoCuishle28**
Signed-off-by: Gaoming <zhanggaoming028@gmail.com>

### What is changed and how it works?

Issue Number: Close #13063

What's Changed:

```commit-message
Adjust the backup organization structure and add a store_id related prefix under the backup path.
```

### Related changes
None

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

#### Manual test
- run `tiup br backup db` to backup the db to S3.
- `drop` the backup db.
- Run `tiup br restore db` to restore the backup on S3.
- Compatibility Test Case 1: Use the pre-modified `br` for backup, and the after-modified `br` for recovery.
- Compatibility Test Case 2: Use the after-modified `br` for backup, and the pre-modified `br` for recovery.

#### Test output
- the directory
Screenshot from 2022-07-05 15-55-44
- run `tiup br restore db` successfully.
- Compatibility Test Case 1 passed.
- Compatibility Test Case 2 passed.

### Release note 

```release-note
In the new backup organization structure, we will see:

./br backup --pd "127.0.0.1:2379" -s "s3://backup/20220621" 
  - After br command finished, we will have the structure below.
➜  backup tree .
.
└── 20220621
    ├── backupmeta
    ├── store1
    │   └── backup-xxx.sst
    ├── store100
    │   └── backup-yyy.sst
    ├── store2
    │   └── backup-zzz.sst
    ├── store3
    ├── store4
    └── store5
```

### CODE REVIEW

File: components/backup/src/endpoint.rs [hunk 1]
```diff
@@ -1089,26 +1092,55 @@ fn get_max_start_key(start_key: Option<&Key>, region: &Region) -> Option<Key> {
...
-            since_the_epoch.as_millis()
-        ),
-        None => format!(
-            "{}_{}_{}",
-            store_id,
-            region.get_id(),
-            region.get_region_epoch().get_version()
-        ),
+        Some(k) => {
+            if storage_name == S3Storage::name() {
```
Comment (overvenus) [inline components/backup/src/endpoint.rs:1107]: Please add some comments and unit tests.

Comment (3pointer) [reply components/backup/src/endpoint.rs:1110]: match (key, storage_name) {
Some(k), S3Storage::name()|LocalStorage::name() => 
Some(k),  _ => 
None, S3Storage::name()|LocalStorage::name() =>
None, _ => 
}

**Comment by MoCuishle28** [comment]
Unit test test_backup_huge_range_and_import failed.
https://github.com/tikv/tikv/blob/59c9676795d08a5ff2e35c899e3c3d30611bd2b9/tests/integrations/backup/mod.rs#L28
Reason: In `fn assert_same_file_name`, two filenames are split using '_' to generate two vectors, and then `assert_eq` each element of these two vectors, except the element at index 4. Because the element at index 4 is timestamp-based.
e.g., `2_1_1_abc_16094_write.sst` -> `[2, 1, 1, abc, 16094, write.sst]`, the skipped element is 16094.

But after adjusting the local backup file organization, the filename becomes `2/1_1_abc_16094_write_sst`. The element at index 4 is `write_sst`, and the timestamp-based element 16094 will not be skipped.

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by overvenus — APPROVED** [review]

**Pull request merged**